feature/quicksettings/src/main/java/com/google/jetpackcamera/feature/quicksettings/QuickSettingsComponents.kt

```diff
@@ -0,0 +1,236 @@
+import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
+import androidx.compose.foundation.lazy.grid.items
+import androidx.compose.foundation.lazy.grid.itemsIndexed
+import androidx.compose.material3.Icon
+import androidx.compose.material3.LocalContentColor
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.draw.scale
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.google.jetpackcamera.quicksettings.R
+import kotlin.math.min
+
+/**
+ * The UI component for quick settings.
+ */
+@Composable
+fun QuickSettingsUi(
+    modifier: Modifier = Modifier,
+    quickSettingsUiModels: List<QuickSettingsUiModel>
+) {
+    var isOpen by remember {
+        mutableStateOf(false)
+    }
+    val backgroundColor = animateColorAsState(
+        targetValue = Color.Black.copy(alpha = if (isOpen) 0.7f else 0f),
+        label = "backgroundColorAnimation"
+    )
+
+    val contentAlpha = animateFloatAsState(
+        targetValue = if (isOpen) 1f else 0f, label = "contentAlphaAnimation",
+        animationSpec = tween()
+    )
+
+    Column(
+        modifier = modifier
+            .fillMaxSize()
+            .background(color = backgroundColor.value),
+        verticalArrangement = Arrangement.Top,
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .wrapContentHeight(),
+            horizontalArrangement = Arrangement.Center,
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Icon(
+                painter = painterResource(R.drawable.baseline_expand_more_72),
+                contentDescription = stringResource(R.string.quick_settings_dropdown_description),
+                tint = if (isOpen) Color.White else LocalContentColor.current,
+                modifier = Modifier
+                    .size(72.dp)
+                    .clickable {
+                        isOpen = !isOpen
+                    }
+                    .scale(1f, if (isOpen) -1f else 1f)
+            )
+        }
+
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .alpha(alpha = contentAlpha.value),
+            verticalArrangement = Arrangement.Center,
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            if (isOpen) {
```

Comment:
I wonder if it's better to place this if block around the column?
That way you only render this if the quick setting is expanded.

Reply:
Done